### PR TITLE
Introduce "too late" presence option to event attendees

### PR DESCRIPTION
**Pull request by ivarnakken**
# Description

[Related backend changes](https://github.com/webkom/lego/pull/3564)

Neither ion-icons or font-awesome had a turtle icon (which I thought was the most fitting), so I extracted one from Lucide (which we're planning on migrating to). I'm very open to design suggestions, as I'm not sure what I feel about this. The padding feels a bit much, so I might refactor the Icon component - but that will probably have to wait.

# Result

If you've made visual changes, please check the boxes below and include images showing the changes. Descriptions are appreciated.

- [x] Changes look good on both light and dark theme.
- [x] Changes look good with different viewports (mobile, tablet, etc.).
- [x] Changes look good with slower Internet connections.

<table>
    <tr>
        <td width="25%">Description</td>
        <td>Before</td>
        <td>After</td>
    </tr>
    <tr>
        <td>
            The icons are much thinner and less "visible", but at the same time they are now Ionicons and therefore match all our other icons - so in that sense they are more "correct".
        </td>
        <td>            
			image
        </td>
        <td>     
			image
        </td>
    </tr>
</table>

The automatic penalty:

image

# Testing

- [x] I have thoroughly tested my changes.

Things work. Tested that all other icons that use the `TooltipIcon` component still work and look good.

### CODE REVIEW

File: app/routes/events/components/EventAdministrate/AttendeeElements.tsx [hunk 1]
```diff
@@ -66,28 +76,63 @@ export const PresenceIcons = ({ presence, registrationId }: PresenceProps) => {
...
-        iconClass={cx('fa fa-check', styles.greenIcon)}
+        iconName="checkmark"
+        iconClass={styles.greenIcon}
         transparent={presence !== 'PRESENT'}
         onClick={() =>
+          eventId &&
           dispatch(updatePresence(eventId, registrationId, 'PRESENT'))
         }
       />
+      <Tooltip content="For sent til stede">
```
Comment (norbye) [inline app/routes/events/components/EventAdministrate/AttendeeElements.tsx:90]:
```suggestion
      <Tooltip content="Møtte opp for sent (gir 1 prikk)">
```

**Review by norbye — APPROVED** [review]
Awesome, love the look!

While the padding may feel like much - I currently like how it is evenly spaced vertically and horizontally. And anyways lets get it out there to get a feel of it

File: app/models.ts [hunk 2]
```diff
@@ -178,7 +178,13 @@ type EventBase = {
 };
 
 export type Permission = string;
-export type EventRegistrationPresence = 'PRESENT' | 'NOT_PRESENT' | 'UNKNOWN';
+
+export type EventRegistrationPresence =
+  | 'PRESENT'
+  | 'TOO_LATE'
+  | 'NOT_PRESENT'
+  | 'UNKNOWN';
```
Comment (eikhr) [inline app/models.ts:186]: enum?

File: app/routes/events/components/EventAdministrate/AttendeeElements.tsx (hunk 1 above)
Comment (eikhr) [reply app/routes/events/components/EventAdministrate/AttendeeElements.tsx:90]: ➕

File: app/routes/events/components/EventAdministrate/AttendeeElements.tsx [hunk 3]
```diff
@@ -66,28 +76,63 @@ export const PresenceIcons = ({ presence, registrationId }: PresenceProps) => {
...
+        iconClass={styles.questionIcon}
+        iconName="help-outline"
         transparent={presence !== 'UNKNOWN'}
         onClick={() =>
+          eventId &&
           dispatch(updatePresence(eventId, registrationId, 'UNKNOWN'))
         }
       />
       <TooltipIcon
         content="Ikke til stede"
```
Comment (eikhr) [inline app/routes/events/components/EventAdministrate/AttendeeElements.tsx:130]:
```suggestion
        content="Ikke til stede (gir 2 prikker)"
```
Just to match @norbye's suggestion

**Review by eikhr — APPROVED** [review]
Awesome!
As for the design, it might also work well with a dropdown like in the bdb

File: app/routes/events/components/EventAdministrate/AttendeeElements.tsx (hunk 3 above)
Comment (ivarnakken) [reply app/routes/events/components/EventAdministrate/AttendeeElements.tsx:130]: Should this say "Møtte ikke opp (gir 2 prikker)" - to match it even further?

**Comment by ivarnakken** [comment]
> As for the design, it might also work well with a dropdown like in the bdb

I contemplated it. Even though it would save us space, it would require the user to click two times, instead of just one, so I ended up not doing it.

File: app/models.ts (hunk 2 above)
Comment (ivarnakken) [reply app/models.ts:186]: Sure

**Review by haukkagu — APPROVED** [review]
Looks good!

**Pull request merged**